src/strategyHelpers/blueGreen/deploy.ts

```diff
@@ -0,0 +1,139 @@
+
+import {Kubectl} from '../../types/kubectl'
+import {RouteStrategy} from '../../types/routeStrategy'
+
+import {
+   BlueGreenDeployment,
+   BlueGreenManifests,
+   deployWithLabel,
+   getManifestObjects,
+   GREEN_LABEL_VALUE,
+   deployObjects
+} from './blueGreenHelper'
+import {setupSMI} from './smiBlueGreenHelper'
+
+import {routeBlueGreenForDeploy} from './route'
+
+export async function deployBlueGreen(
+   kubectl: Kubectl,
+   files: string[],
+   routeStrategy: RouteStrategy
+): Promise<BlueGreenDeployment> {
+   const blueGreenDeployment = await (async () => {
+      switch (routeStrategy) {
+         case RouteStrategy.INGRESS:
+            return await deployBlueGreenIngress(kubectl, files)
+         case RouteStrategy.SMI:
+            return await deployBlueGreenSMI(kubectl, files)
+         default:
+            return await deployBlueGreenService(kubectl, files)
+      }
+   })()
+
+   core.startGroup('Routing blue green')
+   await routeBlueGreenForDeploy(kubectl, files, routeStrategy)
+   core.endGroup()
+   core.debug(
+      `objects deployed for ${routeStrategy}: ${JSON.stringify(
+         blueGreenDeployment.objects
+      )} `
+   )
+   return blueGreenDeployment
+}
+
+export async function deployBlueGreenSMI(
+   kubectl: Kubectl,
+   filePaths: string[]
+): Promise<BlueGreenDeployment> {
+   // get all kubernetes objects defined in manifest files
+   const manifestObjects: BlueGreenManifests = getManifestObjects(filePaths)
+
+   // create services and other objects
+   const newObjectsList = [].concat(
+      manifestObjects.otherObjects,
+      manifestObjects.serviceEntityList,
+      manifestObjects.ingressEntityList,
+      manifestObjects.unroutedServiceEntityList
+   )
+
+   await deployObjects(kubectl, newObjectsList)
+
+   // make extraservices and trafficsplit
+   await setupSMI(kubectl, manifestObjects.serviceEntityList)
+
+   // create new deloyments
+   const blueGreenDeployment: BlueGreenDeployment = await deployWithLabel(
+      kubectl,
+      manifestObjects.deploymentEntityList,
+      GREEN_LABEL_VALUE
+   )
+   return {
+      deployResult: blueGreenDeployment.deployResult,
+      objects: [].concat(blueGreenDeployment.objects, newObjectsList)
+   }
+}
+
+export async function deployBlueGreenIngress(
+   kubectl: Kubectl,
+   filePaths: string[]
+): Promise<BlueGreenDeployment> {
+   // get all kubernetes objects defined in manifest files
+   const manifestObjects: BlueGreenManifests = getManifestObjects(filePaths)
+
+   // create deployments with green label value
+   const servicesAndDeployments = [].concat(
+      manifestObjects.deploymentEntityList,
+      manifestObjects.serviceEntityList
+   )
+   const workloadDeployment: BlueGreenDeployment = await deployWithLabel(
+      kubectl,
+      servicesAndDeployments,
+      GREEN_LABEL_VALUE
+   )
+
+   const otherObjects = [].concat(
+      manifestObjects.otherObjects,
+      manifestObjects.unroutedServiceEntityList
+   )
+
```

Comment:
remove spaces on 99 and 101. Shows that these parts are related

Reply:
I think the spaces preserve readability, is it convention to not have any?

Reply:
spacing out everything doesn't make sense and doesn't help readability

if my code is
```
const apples = getApples()
plantApples(apples)

const grape = getGrape()
plantGrape(grape)
```

it's a lot easier to read than

```
const apples = getApples()

plantApples(apples)

const grape = getGrape()

plantGrape(grape)
```

In this particular case I'd remove the spaces because the logic is grouped. You are getting the objects to deploy then deploying.

Some of it is subjective but spacing everything doesn't make code easier to read because it eliminates the benefit of spaces (breaking up logically grouped parts of code)

Reply:
fair enough, I just thought that none of these three lines are particularly "related" any more than any other collection of lines, plus the fact that prettier makes so many newlines helps separate out these different function calls a bit but it doesn't really matter